--- sys/dev/ata/ata-disk.c	2009-03-13 10:07:05.000000000 -0400
+++ sys/dev/ata/ata-disk.c	2009-04-30 17:13:33.000000000 -0400
@@ -456,7 +456,7 @@
     }
 
     device_printf(dev, "%juMB <%s%s %.8s> at ata%d-%s %s%s\n",
-		  adp->total_secs / (1048576 / DEV_BSIZE),
+		  (adp->total_secs * DEV_BSIZE) / 1000000,
 		  vendor, product, atadev->param.revision,
 		  device_get_unit(ch->dev),
 		  (atadev->unit == ATA_MASTER) ? "master" : "slave",
