--- /root/evoke-head/dists/releng/8.0/src/lib/libc/Versions.def	2009-12-01 11:10:59.000000000 +0000
+++ lib/libc/Versions.def	2010-01-22 03:33:42.000000000 +0000
@@ -9,6 +9,9 @@
 FBSD_1.1 {
 } FBSD_1.0;
 
+EVOKE_0.2 {
+} FBSD_1.1;
+
 # This is our private namespace.  Any global interfaces that are
 # strictly for use only by other FreeBSD applications and libraries
 # are listed here.  We use a separate namespace so we can write
