--- contrib/file/fsmagic.c	2009-04-17 19:17:25.000000000 -0400
+++ contrib/file/fsmagic.c	2009-04-26 01:35:56.000000000 -0400
@@ -95,15 +95,7 @@
 			return -1;
 		return 1;
 	}
-
 	if (mime) {
-		if ((sb->st_mode & S_IFMT) != S_IFREG) {
-			if ((mime & MAGIC_MIME_TYPE) &&
-			    file_printf(ms, "application/x-not-regular-file")
-			    == -1)
-				    return -1;
-			return 1;
-		}
 	}
 	else {
 #ifdef S_ISUID
