diff -ruN sbin/dhclient/dhclient.c sbin/dhclient/dhclient.c
--- sbin/dhclient/dhclient.c	2008-12-22 19:57:06.000000000 -0500
+++ sbin/dhclient/dhclient.c	2009-03-25 17:35:07.000000000 -0400
@@ -2396,6 +2396,8 @@
 	case DHO_DHCP_CLIENT_IDENTIFIER:
 	case DHO_BOOTFILE_NAME:
 	case DHO_DHCP_USER_CLASS_ID:
+	case DHO_TIMEZONE_PCODE:
+	case DHO_TIMEZONE_TCODE:
 	case DHO_END:
 		return (1);
 	case DHO_CLASSLESS_ROUTES:
diff -ruN sbin/dhclient/dhcp.h sbin/dhclient/dhcp.h
--- sbin/dhclient/dhcp.h	2008-12-22 19:57:06.000000000 -0500
+++ sbin/dhclient/dhcp.h	2009-03-25 17:35:59.000000000 -0400
@@ -169,6 +169,8 @@
 #define	DHO_STREETTALK_SERVER		75
 #define	DHO_STREETTALK_DA_SERVER	76
 #define DHO_DHCP_USER_CLASS_ID		77
+#define DHO_TIMEZONE_PCODE		100
+#define DHO_TIMEZONE_TCODE		101
 #define DHO_CLASSLESS_ROUTES		121
 #define DHO_END				255
 
diff -ruN sbin/dhclient/tables.c sbin/dhclient/tables.c
--- sbin/dhclient/tables.c	2008-12-22 19:57:06.000000000 -0500
+++ sbin/dhclient/tables.c	2009-03-25 17:34:10.000000000 -0400
@@ -165,8 +165,8 @@
 	{ "option-97", "X",				&dhcp_universe, 97 },
 	{ "option-98", "X",				&dhcp_universe, 98 },
 	{ "option-99", "X",				&dhcp_universe, 99 },
-	{ "option-100", "X",				&dhcp_universe, 100 },
-	{ "option-101", "X",				&dhcp_universe, 101 },
+	{ "tz-pcode", "t",				&dhcp_universe, 100 },
+	{ "tz-tcode", "t",				&dhcp_universe, 101 },
 	{ "option-102", "X",				&dhcp_universe, 102 },
 	{ "option-103", "X",				&dhcp_universe, 103 },
 	{ "option-104", "X",				&dhcp_universe, 104 },
@@ -400,10 +400,12 @@
 	DHO_IRC_SERVER,
 	DHO_STREETTALK_SERVER,
 	DHO_STREETTALK_DA_SERVER,
+	DHO_TIMEZONE_PCODE,
+	DHO_TIMEZONE_TCODE,
 
 	/* Presently-undefined options... */
 	62, 63, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91,
-	92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105,
+	92, 93, 94, 95, 96, 97, 98, 99, 102, 103, 104, 105,
 	106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117,
 	118, 119, 120, 122, 123, 124, 125, 126, 127, 128, 129, 130,
 	131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142,
