--- lib/libstand/bootp.c	2009-04-17 19:27:19.000000000 -0400
+++ lib/libstand/bootp.c	2009-04-18 16:03:10.000000000 -0400
@@ -582,6 +582,8 @@
 	{85,	__IP,	"nds-servers"},
 	{86,	__TXT,	"nds-tree-name"},
 	{87,	__TXT,	"nds-context"},
+        {100,   __TXT,  "tz-pcode"},
+        {101,   __TXT,  "tz-tcode"},
 	{210,	__TXT,	"authenticate"},
 
 	/* use the following entries for arbitrary variables */
